projects/storefrontlib/cms-components/navigation/search-box/search-box.component.ts

```diff
@@ -379,6 +465,71 @@ export class SearchBoxComponent implements OnInit, OnDestroy {
     }
   }
 
+  // Focus on previous item in results list
+  focusPreviousGroup(event: UIEvent) {
+    event.preventDefault(); // Negate normal keyscroll
+    const [results, focusedGroupIndex] = [
+      this.getGroupElements(),
+      this.getFocusedGroupIndex(),
+    ];
+
+    // Focus on last index moving to first
+    if (results[focusedGroupIndex].length) {
+      if (focusedGroupIndex < 1) {
+        results[0][0].focus();
+      } else {
+        results[focusedGroupIndex - 1][0].focus();
+      }
+    }
+  }
+  // Focus on next item in results list
+  focusNextGroup(event: UIEvent) {
+    this.open();
+    event.preventDefault(); // Negate normal keyscroll
+    const [results, focusedGroupIndex] = [
+      this.getGroupElements(),
+      this.getFocusedGroupIndex(),
+    ];
+    // Focus on first index moving to last
+    if (results.length) {
+      if (focusedGroupIndex >= results.length - 1) {
+        results[0][0].focus();
+      } else {
+        results[focusedGroupIndex + 1][0].focus();
+      }
+    }
+  }
+
+  carouseEventPropagator(event: KeyboardEvent) {
+    if (event) {
```

Comment:
Let's make it simpler and just add a guard before switch statement, so we could avoid too much nesting
```
  if (!event || !event?.code) { 
      return;
    }

    switch (event.code) {
      case 'ArrowRight':
        this.focusNextChild(event);
        return;
      case 'ArrowLeft': {
        this.getGroupElements().forEach((group) => {
          if (group.indexOf(this.getFocusedElement()) !== -1) {
            if (group.indexOf(this.getFocusedElement()) === 0) {
              this.focusPreviousGroup(event);
            } else {
              this.focusPreviousChild(event);
            }
          }
        });

        return;
      }
      case 'ArrowUp':
        this.focusNextGroup(event);
        return;
      default:
        return;
    }
```

Reply:
👍